apiserver/objects.go

```diff
@@ -28,9 +27,9 @@ type ObjectStoreGetter interface {
 }
 
 type objectsCharmHTTPHandler struct {
-	ctxt                httpContext
-	objectStoreGetter   ObjectStoreGetter
-	LegacyCharmsHandler http.Handler
+	ctxt              httpContext
+	objectStoreGetter ObjectStoreGetter
+	LegacyPostHandler FailableHandlerFunc
```

Comment:
FailableHandlerFunc needs renaming, it's generic enough as to be unhelpful to understanding. Why is it Failable? Where is the Infallible version?  What is it handling? 

Perhaps `ObjectHandlerFunc`?

Reply:
`ObjectHandlerFunc` wouldn't be suitable because other handlers use this type. Agree that `Failable` is a red herring, since there is no Unfailable version, but unfortunately it is very generic so I don't think can be improved much

This is it's definition
```
type FailableHandlerFunc func(http.ResponseWriter, *http.Request) error
```

`EndpointMethodHandlerFunc` might be better, since the purpose of this type is to handle a request to a specici endpoint's method? What do you think?

Reply:
EndpointMethodHandlerFunc sounds better than Failable.